gcp_variant_transforms/transforms/merge_headers.py

```diff
@@ -13,26 +13,174 @@
 # limitations under the License.
 
 """beam combiner function for merging VCF file headers."""
-
+from collections import OrderedDict
 import apache_beam as beam
-from gcp_variant_transforms.beam_io import vcf_header_io
 
-__all__ = ['MergeHeaders']
+from gcp_variant_transforms.beam_io.vcf_header_io import VcfHeader
+
+__all__ = ['HeaderConflictResolver', 'HeaderMerger', 'MergeHeaders']
+
+class HeaderConflictResolver(object):
+  """A class for resolving mistmatch between header fields (e.g. type, num).
+  """
+
+  def __init__(self, split_alternate_allele_info_fields=True):
+    """Initialize the class.
+
+    Args:
+     split_alternate_allele_info_fields (bool): if true, all INFO fields with
+        `Number=A` (i.e. one value for each alternate allele) will be stored
+        underthe `alternate_bases` record. If false, they will be stored with
+        the rest of the INFO fields as a repeated fields, hence compatible with
+        `Number=.` for example.
+     """
+    self._split_alternate_allele_info_fields = (
+        split_alternate_allele_info_fields)
+
+  def resolve(self, vcf_field_key, first_vcf_field_value,
+              second_vcf_field_value):
+    """Returns resolution if the conflict between the given field values is
+    resolvable.
+
+    Raises:
+      ValueError: if there is nothing to resolve (values are equal), or
+        the conflict cannot be resolved.
+    Args:
+      vcf_field_key (string): field key in VCF header.
+      first_vcf_field_value (integer or string): first field value.
+      second_vcf_field_value (integer or string): second field value.
+   """
+    if vcf_field_key == 'type':
+      return self._resolve_type(first_vcf_field_value, second_vcf_field_value)
+    elif vcf_field_key == 'num':
+      return self._resolve_number(first_vcf_field_value, second_vcf_field_value)
+    else:
+      # We only care about conflicts in 'num' and 'type' fields.
+      return first_vcf_field_value
+
+  def _resolve_type(self, first, second):
+    if first == second:
+      raise ValueError('Types are the same. Nothing to resolve: {}, {}'
```

Comment:
Is it better to just return first in that case? I think this is semantically ok and we can even simplify the code below to always call resolve without the additional if statement. It may be a bit less efficient, but may be simpler.

Reply:
Done